tests/cli/test_compose_alibaba.sh

```diff
@@ -0,0 +1,197 @@
+        rlRun -t -c "$ALI_DIR/aliyun ecs ImportKeyPair --KeyPairName $KEY_NAME --PublicKeyBody '$SSH_PUB_KEY'"
+
+        RELEASE_TIME=$(date -u -d "24 hours" '+%FT%TZ')
+
+        # SecurityGroup is composer-allow-ssh
+        # VPC is composer-vpc
+        response=$($ALI_DIR/aliyun ecs RunInstances --Amount 1 --ImageId $IMAGE_ID \
+                    --InstanceType=$INSTANCE_TYPE --InstanceName Composer-Test-VM-$UUID \
+                    --SecurityGroupId sg-0xi4w9isg0p1ytj1qbhf \
+                    --VSwitchId vsw-0xi36w0a9l894vf2momfb \
+                    --KeyPairName $KEY_NAME \
+                    --InternetMaxBandwidthIn 5 --InternetMaxBandwidthOut 5 \
+                    --AutoReleaseTime $RELEASE_TIME)
+        rlAssert0 "VM started successfully" $?
+        rlLogInfo "$response"
+
+        INSTANCE_ID=`echo "$response" | jq .InstanceIdSets.InstanceIdSet | jq -r '.[0]' `
+
+        until [ $($ALI_DIR/aliyun ecs DescribeInstanceAttribute --InstanceId $INSTANCE_ID | jq -r .Status | grep "Running\|Stopped") ]; do
+            sleep 30
+            rlLogInfo "Waiting for instance to start ..."
+        done
+
+        rlAssertEquals "Instance $INSTANCE_ID is Running" \
+            "$($ALI_DIR/aliyun ecs DescribeInstanceAttribute --InstanceId $INSTANCE_ID | jq -r .Status)" "Running"
+        rlRun -t -c "$ALI_DIR/aliyun ecs DescribeInstanceAttribute --InstanceId $INSTANCE_ID"
+
+        IP_ADDRESS="null"
+        while [ "$IP_ADDRESS" == "null" ]; do
+            rlLogInfo "IP_ADDRESS is not assigned yet ..."
+            sleep 30
+            IP_ADDRESS=`$ALI_DIR/aliyun ecs DescribeInstanceAttribute --InstanceId $INSTANCE_ID | jq -r .PublicIpAddress.IpAddress | jq -r '.[0]'`
+        done
+
+        rlLogInfo "Running INSTANCE_ID=$INSTANCE_ID with IP_ADDRESS=$IP_ADDRESS"
+    rlPhaseEnd
+
+    rlPhaseStartTest "Verify EC2 instance"
+        # cloud-init default config differs between RHEL and Fedora
+        CLOUD_USER="cloud-user"
+        if [ -f "/etc/fedora-release" ]; then
+            CLOUD_USER="fedora"
+        fi
+
+        # run generic tests to verify the instance
+        verify_image "$CLOUD_USER" "$IP_ADDRESS" "-i $SSH_KEY_DIR/id_rsa"
+    rlPhaseEnd
+
+    rlPhaseStartCleanup
+        rlRun -t -c "$ALI_DIR/aliyun ecs DeleteInstance --Force True --InstanceId $INSTANCE_ID"
+        rlRun -t -c "$ALI_DIR/aliyun ecs DeleteImage --Force True --ImageId $IMAGE_ID"
+        rlRun -t -c "$ALI_DIR/aliyun oss rm oss://$ALICLOUD_BUCKET/$IMAGE --force"
+        rlRun -t -c "$CLI compose delete $UUID"
+        rlRun -t -c "rm -rf $IMAGE $SSH_KEY_DIR $ALI_DIR"
+        # do this here to give time for the VM instance to be removed properly
+        # also don't fail if the key is still attached to an instance which is waiting
+        # to be desroyed. We're going to remove these keys in cleanup afterwards
+        $ALI_DIR/aliyun ecs DeleteKeyPairs --KeyPairNames "['$KEY_NAME']" || echo
```

Comment:
I don't understand what line 193 does. The `$ALI_DIR` is removed on line 189, so can this ever do something?

Reply:
This deletes keypairs from the Alicloud console, not from the local filesystem.

Reply:
Ok... I don't know how the Alicloud console works, I just saw this `/tests/cli/test_compose_alibaba.sh: line 193: /tmp/alicloud.c6k0i/aliyun: No such file or directory` in the log and thought that it does nothing.

Reply:
Crap, that is a bug. Good catch. This line should be move 1 line up.